lib/gooddata/lcm/helpers/check_helper.rb

```diff
@@ -18,20 +18,32 @@ def check_params(specification, params)
               if specification[param_name][:opts][:default]
                 params[param_name] = specification[param_name][:opts][:default]
               elsif specification[param_name][:opts][:required]
-                fail("Mandatory parameter '#{param_name}' of type '#{type}' is not specified")
+                if ENV['RSPEC_ENV'] == 'test'
+                  fail "Mandatory parameter '#{param_name}' of type '#{type}' is not specified"
+                else
+                  GoodData.logger.error("Mandatory parameter '#{param_name}' of type '#{type}' is not specified")
+                end
               end
             else
               if type.class.const_get(:CATEGORY) == :complex && !value.is_a?(Hash)
                 puts JSON.pretty_generate(params)
-                fail "Expected parameter '#{param_name}' to be kind of '#{type}', got '#{value.class.name}'"
+                if ENV['RSPEC_ENV'] == 'test'
+                  fail "Expected parameter '#{param_name}' to be kind of '#{type}', got '#{value.class.name}'"
```

Comment:
Refers to 'value' more than self (maybe move it to another class?) (FeatureEnvy)